hyundai_kia_connect_api/ApiImpl.py

```diff
@@ -1,75 +1,75 @@
 import datetime as dt
 import logging
+import re
 from dataclasses import dataclass
 
 import requests
 
 from .const import *
 from .Token import Token
-from .Vehicle import Vehicle, EvChargeLimits
+from .Vehicle import Vehicle, EvChargeLimits, ClimateRequestOptions
 
 from .utils import (
     get_child_value,
-    get_hex_temp_into_index,
-    get_index_into_hex_temp,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
 
-
-@dataclass
-class ClimateRequestOptions:
-    set_temp: float = None
-    duration: int = None
-    defrost: bool = None
-    climate: bool = None
-    heating: int = None
-    front_left_seat: int = None
-    front_right_seat: int = None
-    rear_left_seat: int = None
-    rear_right_seat: int = None
-
 class ApiImpl:
     data_timezone = dt.timezone.utc
     temperature_range = None
 
     def __init__(self) -> None:
         """Initialize."""
+        self.token: Token = None
 
-    def login(self, username: str, password: str) -> Token:
+    def login(self, username: str, password: str, pin: str):
         """Login into cloud endpoints and return Token"""
         pass
 
-    def get_vehicles(self, token: Token) -> list[Vehicle]:
+    def get_vehicles(self) -> list[Vehicle]:
```

Comment:
the Token attribute is now part of the ApiImpl class so no longer needed as a method argument (can be accessed directly by the object methods)